secret_api.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"net"
+	"net/http"
+)
+
+type BaconApiController struct {
+	Farm *Farm
+	sock net.Conn // shhh, private!
+}
+
+func NewBaconAPIController(farm *Farm) (*BaconApiController, error) {
+	s, err := net.Dial("tcp", os.Getenv("SECRET_TCP_HOST"))
```

Comment:
You're leaking a socket.

![blobfish](https://cloud.githubusercontent.com/assets/571786/11467350/d1c53450-9747-11e5-855a-9c676ebba412.jpg)